units/src/fee.rs

```diff
@@ -146,19 +149,24 @@ impl FeeRate {
     /// fee is a non-integer amount, the amount is rounded up, ensuring that the transaction fee is
     /// enough instead of falling short if rounded down.
     ///
-    /// Returns [`None`] if overflow occurred.
+    /// Returns [`NumOpResult::Error`] if overflow occurred.
     #[must_use]
-    pub const fn checked_mul_by_weight(self, weight: Weight) -> Option<Amount> {
-        // No `?` operator in const context.
+    pub const fn checked_mul_by_weight(self, weight: Weight) -> NumOpResult<Amount> {
```

Comment:
Oh, this is annoying! I think `checked_` methods should return `Option` to be consistent with integers but this is a building block for other functions...

Reply:
works for me.